plenum/server/view_change/view_changer.py

```diff
@@ -517,20 +526,31 @@ def startViewChange(self, proposed_view_no: int):
         # TODO: view change is a special case, which can have different
         # implementations - we need to make this logic pluggable
 
-        for view_no in tuple(self._next_view_indications.keys()):
-            if view_no > proposed_view_no:
-                break
-            self._next_view_indications.pop(view_no)
-
         self.view_no = proposed_view_no
         self.view_change_in_progress = True
         self.previous_master_primary = self.node.master_primary_name
         self.set_defaults()
+        self._process_vcd_for_future_view()
+
         self.initInsChngThrottling()
 
         self.node.on_view_change_start()
         self.node.start_catchup()
 
+    def _process_vcd_for_future_view(self):
+        # make sure that all received VCD messages for future view
+        # (including the current view) are stored, as they will be needed for a quorum
+        # to finish the View Change and start selection.
+        # This is especially critical for Propagate Primary mode (on receiving CURRENT_STATE by a new node).
+        if self.view_no in self._next_view_indications:
+            for frm, vcd in self._next_view_indications[self.view_no].items():
+                self._on_verified_view_change_done_msg(vcd, frm)
```

Comment:
Probably it's better to call `process_vchd_msg` here, which will call `_on_verified_view_change_done_msg` only when it's truly verified?

Reply:
It will not work. Added a comment with a description.